src/lib/sw_engine/tvgSwCommon.h

```diff
@@ -278,6 +278,21 @@ static inline uint32_t COLOR_INTERPOLATE(uint32_t c1, uint32_t a1, uint32_t c2,
     return (c1 |= t);
 }
 
+//c1(x,y)--a1-- c2
+// |        |    |
+// |        |    |
+//a2---result---a2
+// |        |    |
+// |        |    |
+//c4 ------a1-- c3
+static inline uint32_t COLOR_BILINEAR_INTERPOLATE(uint32_t c1, uint32_t c2, uint32_t c3, uint32_t c4, uint32_t a1, uint32_t a2)
+{
+    return COLOR_INTERPOLATE(COLOR_INTERPOLATE(c1, 255 - a1, c2, a1),
+                             255 - a2,
+                             COLOR_INTERPOLATE(c4, 255 - a1, c3, a1),
+                             a2);
```

Comment:
I'm not sure about the format rules that we have - should we use one line? or can we split?
also the comment is a bit confusing for me, but as you wish :)

Reply:
I modified the code. One line is recommended. But... for readability, I think newlines can be allowed. (if the code line is toooooo long)